pkg/operator/ceph/csi/spec.go

```diff
@@ -313,3 +322,53 @@ func StartCSIDrivers(namespace string, clientset kubernetes.Interface, ver *vers
 	}
 	return nil
 }
+
+// ValidateCSIVersion checks if the configured ceph-csi image is supported
+func ValidateCSIVersion(clientset kubernetes.Interface, namespace, csiPluginImage, rookImage, serviceAccountName string) error {
+	// skip check if running unsupported images is allowed
+	if CSIParam.AllowUnsupported {
+		logger.Infof("Allowing unsupported ceph-csi images -> no version check")
+		return nil
+	}
+
+	ownerRef := metav1.OwnerReference{
+		APIVersion: fmt.Sprintf("%s", "detect-csi-version"),
```

Comment:
Is this owner reference valid? It must be a real resource. I'd suggest using the operator deployment as the owner reference.